php-checks/src/main/java/org/sonar/php/checks/security/SessionFixationStrategyCheck.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks.security;
+
+import java.util.Optional;
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.CheckUtils;
+import org.sonar.plugins.php.api.tree.Tree.Kind;
+import org.sonar.plugins.php.api.tree.declaration.CallArgumentTree;
+import org.sonar.plugins.php.api.tree.expression.ArrayInitializerTree;
+import org.sonar.plugins.php.api.tree.expression.ArrayPairTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+import org.sonar.plugins.php.api.tree.expression.LiteralTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+import static org.sonar.php.checks.utils.CheckUtils.argument;
+import static org.sonar.php.checks.utils.CheckUtils.lowerCaseFunctionName;
+import static org.sonar.php.checks.utils.CheckUtils.trimQuotes;
+
+@Rule(key = "S5876")
+public class SessionFixationStrategyCheck extends PHPVisitorCheck {
+
+  private static final String MESSAGE = "Create a new session during user authentication to prevent session fixation attacks.";
+  private static final String SENSITIVE_VALUE = "none";
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    String functionName = lowerCaseFunctionName(tree);
+    if (("loadfromextension".equals(functionName) && isArgumentEqualsTo(tree, "extension", 0, "security") && isArgumentSensitiveArray(tree, "values", 1)) ||
```

Comment:
https://jira.sonarsource.com/browse/RSPEC-6090 was updated. We should now raise on `extension` rather than `loadfromextension`.

Reply:
Now the test case is working code, but does not reflect the documentation. We should still handle loadFromExtension additionally

Reply:
I added extension.